frontend/javascripts/components/slider.tsx

```diff
@@ -0,0 +1,100 @@
+import { Slider as AntdSlider, type SliderSingleProps } from "antd";
+import type { SliderRangeProps } from "antd/lib/slider";
+import type { WheelEventHandler } from "react";
+
+const DEFAULT_WHEEL_FACTOR = 0.02;
+const DEFAULT_STEP = 1;
+
+type SliderProps = (SliderSingleProps | SliderRangeProps) & {
+  wheelFactor?: number;
+  disableOnWheel?: boolean;
+};
+
+const getDiffPerSliderStep = (
+  sliderRange: number,
+  factor: number | undefined = DEFAULT_WHEEL_FACTOR,
+  step: number | undefined | null,
+) => {
+  let stepNotNull = step || DEFAULT_STEP;
+  let result = (factor || DEFAULT_WHEEL_FACTOR) * sliderRange;
+  if (result < stepNotNull) return stepNotNull;
+  return result;
+};
+
+const getWheelStepFromEvent = (
+  step: number | undefined | null,
+  deltaY: number,
+  wheelStep: number,
+) => {
+  // Make sure that result is a multiple of step
+  return (
+    (step || DEFAULT_STEP) *
+    Math.round((wheelStep * deltaY) / Math.abs(deltaY) / (step || DEFAULT_STEP))
+  );
+};
+
+export function Slider(props: SliderProps) {
+  const {
+    min,
+    max,
+    onChange,
+    value,
+    range,
+    defaultValue,
+    wheelFactor,
+    disableOnWheel,
+    step,
+    disabled,
+  } = props;
+  if (min == null || max == null || onChange == null || value == null || disabled)
+    return <AntdSlider {...props} />;
+  const sliderRange = max - min;
+  let handleWheelEvent: WheelEventHandler<HTMLDivElement> = () => {};
+  let handleDoubleClick: React.MouseEventHandler<HTMLDivElement> = () => {};
+  const wheelStep = getDiffPerSliderStep(sliderRange, wheelFactor, step);
+  // diffentiate between single value and range slider
+  if (range === false || range == null) {
+    if (!disableOnWheel) {
+      handleWheelEvent = (event) => {
+        const newValue = value - getWheelStepFromEvent(step, event.deltaY, wheelStep);
+        if (newValue < min) onChange(min);
+        else if (newValue > max) onChange(max);
+        else onChange(newValue);
+      };
+    }
+    // This code is duplicated because TypeScript doesn't understand that onChange
+    // always takes the type of defaultValue.
+    handleDoubleClick = (event) => {
+      if (event.target instanceof HTMLElement) {
+        if (event.target.className.includes("ant-slider-handle")) {
+          if (defaultValue != null) onChange(defaultValue);
+        }
+      }
+    };
+  } else if (range === true || typeof range === "object") {
+    if (!disableOnWheel) {
+      handleWheelEvent = (event) => {
+        const newMin = Math.round(
+          value[0] + getDiffPerSliderStep(event.deltaY, sliderRange, wheelFactor),
+        );
+        const newMax = Math.round(
+          value[1] - getDiffPerSliderStep(event.deltaY, sliderRange, wheelFactor),
+        );
+        if (newMin >= min && newMax <= max && min < max) onChange([newMin, newMax]);
```

Comment:
I'd say this code behaves unexpected the following scenario:
If the range is `[0,100] ` and min is currently at 2 and max at 50. How I scroll to increase the range which would result in `newMin = -2` and `newMax = 54` (assuming `getDiffPerSliderStep(event.deltaY, sliderRange, wheelFactor)` returns 4).
But the ` if (newMin >= min && newMax <= max && min < max)` condition would ignore this update.

I would expect that in this scenario the range is updated to `min = 0, max = 54` -> clamping to the min and max boards.

Therefore I would suggest something more like:

```suggestion
        let newMin = Math.round(
          value[0] + getDiffPerSliderStep(event.deltaY, sliderRange, wheelFactor),
        );
        let newMax = Math.round(
          value[1] - getDiffPerSliderStep(event.deltaY, sliderRange, wheelFactor),
        );
        newMin = clamp(min, newMin, Math.min(newMax, max));
        newMax = clamp(newMin, newMax, max);
        onChange([newMin, newMax]);
```

Reply:
I feel like this is more a design decision than the cause for the error :) The error was caused by me calling the wrong method :sweat_smile:

Reply:
anyway I think this makes total sense and I adopted the code (essentially) and the behavior!

Reply:
... and hopefully fixed the weird behaviour :) thank you for the thorough testing!

Reply:
jup its gone, cannot reproduce it anymore :rocket: